addons/mrp/models/stock_move.py

```diff
@@ -97,10 +97,24 @@ def _get_aggregated_product_quantities(self, **kwargs):
         """
         aggregated_move_lines = super()._get_aggregated_product_quantities(**kwargs)
         kit_name = kwargs.get('kit_name')
-        if kit_name:
-            for aggregated_move_line in aggregated_move_lines:
-                if aggregated_move_lines[aggregated_move_line]['description'] == kit_name:
+
+        to_be_removed = []
+        for aggregated_move_line in aggregated_move_lines:
+            bom_lines = aggregated_move_lines[aggregated_move_line]['product'].bom_line_ids
+            bom_type = bom_lines[0].bom_id.type if bom_lines else ""
```

Comment:
I guess this could be shortened to:

```suggestion
            is_phantom = bom_lines[:1].bom_id.type == 'phantom'
```